infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/CorfuLogReplicationRuntime.java

```diff
@@ -162,14 +164,18 @@ public class CorfuLogReplicationRuntime {
     /**
      * Default Constructor
      */
-    public CorfuLogReplicationRuntime(LogReplicationRuntimeParameters parameters, LogReplicationMetadataManager metadataManager,
-        LogReplicationConfigManager replicationConfigManager, ReplicationSession replicationSession) {
+    public CorfuLogReplicationRuntime(LogReplicationRuntimeParameters parameters,
+                                      LogReplicationMetadataManager metadataManager, UpgradeManager upgradeManager,
+                                      LogReplicationConfigManager replicationConfigManager,
+                                      ReplicationSession replicationSession) {
         this.remoteClusterId = replicationSession.getRemoteClusterId();
         this.metadataManager = metadataManager;
+        this.upgradeManager = upgradeManager;
```

Comment:
we shouldn't need to keep this as a variable of Runtime as it's passed in iniinitializeStates down to negotiatingState

Reply:
yes, but it was created before, right? I'm just saying you can pass upgradeManager directly in initializeStates() and not keep it as a class variable which is never used again.

Reply:
got it.

It is actually used twice right now - during start and on sink cluster change.  But it can be passed to those methods.

Since the place of creating it is changing anyway with Chris's PR, shall I leave it like this for now?

Reply:
Sure I could include that change, pass it directly to initialize states instead of creating a local variable